generators/server/templates/src/main/java/package/config/_LoggingConfiguration.java

```diff
@@ -84,16 +88,22 @@ public void addLogstashAppender(LoggerContext context) {
         String customFields = "{\"app_name\":\"" + appName + "\",\"app_port\":\"" + serverPort + "\"}";
         <%_ } _%>
 
+        //More documentation is available at: https://github.com/logstash/logstash-logback-encoder
+        LogstashEncoder logstashEncoder=new LogstashEncoder();
         // Set the Logstash appender config from JHipster properties
-        logstashAppender.setSyslogHost(jHipsterProperties.getLogging().getLogstash().getHost());
-        logstashAppender.setPort(jHipsterProperties.getLogging().getLogstash().getPort());
-        logstashAppender.setCustomFields(customFields);
-
-        // Limit the maximum length of the forwarded stacktrace so that it won't exceed the 8KB UDP limit of logstash
+        logstashEncoder.setCustomFields(customFields);
+        // Set the Logstash appender config from JHipster properties
+        logstashAppender.addDestinations(new InetSocketAddress(jHipsterProperties.getLogging().getLogstash().getHost(),jHipsterProperties.getLogging().getLogstash().getPort()));
+        
         ShortenedThrowableConverter throwableConverter = new ShortenedThrowableConverter();
-        throwableConverter.setMaxLength(7500);
+        throwableConverter.setMaxDepthPerThrowable(30);
```

Comment:
I'm not so keen on magic numbers. Please add config properties in https://github.com/jhipster/jhipster/blob/master/src/main/java/io/github/jhipster/config/JHipsterProperties.java#L797
by doing a PR on our server lib: https://github.com/jhipster/jhipster
Please add the following under `jhipster.logging.logstash`:
- maxDepthPerThrowable (default value 30)
- keepAliveDurationMinutes (default 5)
- reconnectionDelaySeconds (default 1)
Also you need to add those to application-dev.yml and application-prod.yml so that user can easily discover and customize those values.